src/firefly/js/visualize/ChartsTableViewPanel.jsx

```diff
@@ -527,7 +589,7 @@ export class ChartsTableViewPanel extends Component {
     }
 
     getNextState() {
-        const tblId = TblUtil.getActiveTableId();
+        const tblId = this.props.tblId || TblUtil.getActiveTableId();
```

Comment:
this may be an issue.  TblUtil.getActiveTableId() returns the active table for 'main' group.  if this chart is connected to a different group, it may switches to an unintended table.